third_party/ijar/test/zip_test.sh

```diff
@@ -188,4 +188,22 @@ function test_zipper_specify_path() {
       || fail "Unzip after zipper output is not expected"
 }
 
+function test_zipper_permissions() {
+  mkdir -p ${TEST_TMPDIR}/files
+  printf "#!/bin/sh\nexit 0\n" > ${TEST_TMPDIR}/files/executable
+  printf "#!/bin/sh\nexit 0\n" > ${TEST_TMPDIR}/files/non_executable
+  chmod +x ${TEST_TMPDIR}/files/executable
+  chmod -x ${TEST_TMPDIR}/files/non_executable
+
+  ${ZIPPER} cC ${TEST_TMPDIR}/output.zip \
+      executable=${TEST_TMPDIR}/files/executable \
+      non_executable=${TEST_TMPDIR}/files/non_executable
+
+  mkdir -p ${TEST_TMPDIR}/out
+  (cd ${TEST_TMPDIR}/out && $UNZIP -q ${TEST_TMPDIR}/output.zip)
+
+  test -x ${TEST_TMPDIR}/out/executable || fail "out/executable should have been executable"
+  !(test -x ${TEST_TMPDIR}/out/non_executable) || fail "out/non_executable should not have been executable"
```

Comment:
Why use a subshell? No need for parentheses here.

For better readability, I recommend if-then rather than the short-circuiting `||`:
```
if ! test -x ${TEST_TMPDIR}/out/non_executable; then
  fail "out/non_executable should not have been executable"
fi
```

Reply:
Done